cmake/scripts/buildinfo.cmake

```diff
@@ -19,20 +19,31 @@ if (NOT ETH_BUILD_PLATFORM)
 	set(ETH_BUILD_PLATFORM "unknown")
 endif()
 
-# Logic here: If prerelease.txt exists but is empty, it is a non-pre release.
-# If it does not exist, create our own prerelease string
+
 if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
-	file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
-	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
-else()
+	message(FATAL_ERROR "The usage of the file prerelease.txt is no longer supported. Please use SOL_FORCE_RELEASE option if you want to build a release executable or SOL_PRERELEASE_STRING if you want to adjust the pre-release part of the version string.")
+endif()
+
+if (SOL_FORCE_RELEASE)
```

Comment:
Shouldn't `SOL_FORCE_RELEASE` and `SOL_VERSION_PRERELEASE` be somewhere defined as an CMake [`option`](https://cmake.org/cmake/help/latest/command/option.html)?

Reply:
Yeah, I think they should. Good point.

Reply:
@nishant-sachdeva 
> Could you add some detail to https://github.com/ethereum/solidity/pull/12717#discussion_r834722129
> I'm not very well versed with cmake scripting and am not very sure what you mean by "adding the options"

See how we use it here: https://github.com/ethereum/solidity/blob/c4909e99c1015233d6937d229058cde52615f246/CMakeLists.txt#L35-L37

And also see [docs for `option`](https://cmake.org/cmake/help/latest/command/option.html).

Reply:
@cameel exactly. @nishant-sachdeva basically just define `SOL_FORCE_RELEASE` and `SOL_VERSION_PRERELEASE` as an option in `solidity/CMakeLists.txt`.